### PR TITLE
JAMES-2884 Improve & refresh JMAP documentation

### PR DESCRIPTION
**Pull request by chibenwa**
Screenshot from 2020-11-21 11-14-42

### CODE REVIEW

File: server/protocols/jmap-rfc-8621/doc/specs/spec/mail/intro.mdown [hunk 1]
```diff
@@ -50,33 +50,37 @@ This document defines three additional capability URIs.
 
 ### urn:ietf:params:jmap:mail
 
-<aside class="notice">
-  Implemented
-</aside>
+
+> :warning:
+> Implemented
```
Comment (Arsnael) [inline server/protocols/jmap-rfc-8621/doc/specs/spec/mail/intro.mdown:55]: A warning for implemented stuff?

File: server/protocols/jmap-rfc-8621/doc/specs/spec/mail/intro.mdown [hunk 2]
```diff
@@ -123,9 +126,8 @@ The value of this property in an account's *accountCapabilities* property is an
 
 ### urn:ietf:params:jmap:vacationresponse
 
-<aside class="warning">
-  Not implemented yet
-</aside>
+> :warning:
+> Implemented
```
Comment (Arsnael) [inline server/protocols/jmap-rfc-8621/doc/specs/spec/mail/intro.mdown:130]: idem

File: server/protocols/jmap-rfc-8621/doc/specs/spec/mail/mailbox.mdown [hunk 3]
```diff
@@ -5,26 +5,27 @@ A Mailbox represents a named set of Emails. This is the primary mechanism for or
 For compatibility with IMAP, an Email MUST belong to one or more Mailboxes. The Email id does not change if the Email changes Mailboxes.
 
 A **Mailbox** object has the following properties:
-<aside class="notice">
-  Implemented
-</aside>
+
+> :warning:
+> Implemented
```
Comment (Arsnael) [inline server/protocols/jmap-rfc-8621/doc/specs/spec/mail/mailbox.mdown:10]: idem

File: server/protocols/jmap-rfc-8621/doc/specs/spec/mail/mailbox.mdown [hunk 4]
```diff
@@ -105,9 +109,8 @@ The following JMAP methods are supported.
 
 ## Mailbox/get
 
-<aside class="notice">
-  Implemented
-</aside>
+> :warning:
+> Implemented
```
Comment (Arsnael) [inline server/protocols/jmap-rfc-8621/doc/specs/spec/mail/mailbox.mdown:113]: idem

File: server/protocols/jmap-rfc-8621/doc/specs/spec/mail/mailbox.mdown [hunk 5]
```diff
@@ -174,9 +176,8 @@ This is a standard "/queryChanges" method as described in [@!RFC8620], Section 5
 
 ## Mailbox/set
 
-<aside class="notice">
-  Implemented
-</aside>
+> :warning:
+> Implemented
```
Comment (Arsnael) [inline server/protocols/jmap-rfc-8621/doc/specs/spec/mail/mailbox.mdown:180]: idem

File: server/protocols/jmap-rfc-8621/doc/specs/spec/mail/messagesubmission.mdown [hunk 6]
```diff
@@ -187,13 +202,24 @@ This is a standard "/queryChanges" method as described in [@!RFC8620], Section 5
...
+> Partially implemented. Only EmailSubmission/set create is supported.
+> No result is stored.
+
 This is a standard "/set" method as described in [@!RFC8620], Section 5.3 with the following two additional request arguments:
 
 - **onSuccessUpdateEmail**: `Id[PatchObject]|null`
   A map of *EmailSubmission id* to an object containing properties to update on the Email object referenced by the EmailSubmission if the create/update/destroy succeeds. (For references to EmailSubmissions created in the same "/set" invocation, this is equivalent to a creation-reference, so the id will be the creation id prefixed with a `#`.)
+
+> :warning:
+> Implemented
```
Comment (Arsnael) [inline server/protocols/jmap-rfc-8621/doc/specs/spec/mail/messagesubmission.mdown:215]: idem

File: server/protocols/jmap-rfc-8621/doc/specs/spec/mail/messagesubmission.mdown [hunk 7]
```diff
@@ -187,13 +202,24 @@ This is a standard "/queryChanges" method as described in [@!RFC8620], Section 5
...
+
 - **onSuccessDestroyEmail**: `Id[]|null`
   A list of *EmailSubmission ids* for which the Email with the corresponding emailId should be destroyed if the create/update/destroy succeeds. (For references to EmailSubmission creations, this is equivalent to a creation-reference so the id will be the creation id prefixed with a `#`.)
 
+> :warning:
+> Implemented
```
Comment (Arsnael) [inline server/protocols/jmap-rfc-8621/doc/specs/spec/mail/messagesubmission.mdown:221]: idem

File: server/protocols/jmap-rfc-8621/doc/specs/spec/mail/vacationresponse.mdown [hunk 8]
```diff
@@ -1,8 +1,7 @@
 # Vacation Response
 
-<aside class="warning">
-  Partially implemented
-</aside>
+> :warning:
+> Implemented
```
Comment (Arsnael) [inline server/protocols/jmap-rfc-8621/doc/specs/spec/mail/vacationresponse.mdown:4]: idem

File: server/protocols/jmap-rfc-8621/doc/specs/spec/mail/intro.mdown (hunk 1 above)
Comment (chibenwa) [reply server/protocols/jmap-rfc-8621/doc/specs/spec/mail/intro.mdown:54]: Using :information_source: instead...

**Review by LanKhuat — APPROVED** [review]

**Review by Arsnael — APPROVED** [review]

**Comment by chibenwa** [comment]
Merged

**Pull request closed without merging**